### DISCUSSION TITLE
RootModel with Pylance/Pyright: "Expected 0 positional Arguments"

### DISCUSSION THREAD

**Discussion by RS-Credentive** [pydantic/pydantic#6976] [Question]
All, I am pulling my hair out here. I make heavy use of RootModels within my project, and Pylance is barfing all over my code:

Assume a Root model defined as follows:
```py
class Token(RootModel[str]):
    """
    A class representing a non-colonized name as defined by XML Schema Part 2: Datatypes Second Edition.

    Attributes:
        root: a non-colonized name as defined by XML Schema Part 2: Datatypes Second Edition.
    """

    root: str = Field(
        description="Non-colonized token type used for various identifiers",
        pattern=r"^([a-zA-Z_])([A-Za-z0-9.\-_])*$",  # any non-numeric character or "_", followed by a sequence of any alphanumeric character, "_", "-", or "."
    )
```

If we initialize an instance like this:
```py
datatypes.Token("type")
```

Pylance/pyright give me the following error: "Expected 0 positional Arguments"

It works fine when executed - but I'm concerned that I have defined something the wrong way and some subtle bug is going to bite me in the ass. Anybody seen this before? I can't find any reference to it anywhere online.

**Comment by ja-albert** [comment]
I face the same problem with `mypy`. Have you found a solution or workaround?

**Comment by RS-Credentive** [comment]
Not yet.

**Comment by brendanmaguire** [comment]
```python
datatypes.Token(root="type")
```
This will work. It's not pretty. I came here looking for a nicer way to do this.

**Comment by madkopp** [comment]
Having the same issue.  However, if I just define __init__ in the child class and just call super, it fixes the issue.  It seems that the problem is that Pylance/Pyright don't recognize the __init__ functions of parent classes.  In the example below, if I don't put the "useless" __init__ definition in ClassB, Pylance gives me an "Expected 0 positional arguments" message.

```
"""
Minimal code to reproduce Pydantic/Pylance inheritance issue.
"""

from typing import Any
from pydantic import BaseModel


class ClassA(BaseModel):
    """Parent class."""

    var_a: str = "a"
    var_b: list[str]

    def __init__(self, *items: str, **data: Any) -> None:
        """Custom init function that includes positional args."""
        data["var_b"] = [item for item in items]
        super().__init__(**data)


class ClassB(ClassA):
    """Child class."""

    var_a: str = "b"

    def __init__(self, *items: str, **data: Any) -> None:
        """Useless init definition fixes static typing issue."""
        super().__init__(*items, **data)


ClassB("a", "b")
```

**Comment by camilaagw** [comment]
Having the same issue when I create a Pydantic model with `from_attributes=True`
